### ISSUE TITLE
progressbar disappears when pre-animate it (bug)

### ISSUE THREAD

**Issue by UtkuDalmaz** [drawRect/Instagram_Stories#89]
When I pre-animate (start animation but don't complete it) the snaps, the progress indicator disappears unexpectedly

Please check the video

https://drive.google.com/open?id=1sjxeNecqFOz-M8tkHzRaMMFW4ETLNiw-

**Comment by ranmyfriend** [comment]
Hi @UtkuDalmaz 
Thanks for your attachment. Please give me sometime,. let me add more videos and play with the issue and will give you the fix

**Comment by UtkuDalmaz** [comment]
@ranmyfriend Ok man thank you.

**Comment by ranmyfriend** [comment]
Hi @UtkuDalmaz 
Just now I have tested **it works** fine from my side.
I have used iOS 13.3.1 real device and also tested with the simulator on the latest
The same scenario(1 video) I have tried it here. Can you please let me know the exact scenario or give me more info about the bug. So we can reproduce it and give you some fix

**Comment by UtkuDalmaz** [comment]
I realized that it only happens in the simulator.  Sorry for the false alarm.

**Comment by UtkuDalmaz** [comment]
Can you help me on this, how can I call "resumeEntireSnap()" from IGStoryPreviewController ? 
I mean how can I resume snap from IGStoryPreviewController?

**Comment by ranmyfriend** [comment]
@UtkuDalmaz 
**resumeEntireSnap** you can call from StoryPreviewCell based on the IndexPath. So you must have the index path in StoryPreviewController. then try to get the cell using cellForRowAtIndexpath then simply call - cell.resumeEntireSnap()

**Comment by UtkuDalmaz** [comment]
@ranmyfriend Man sorry but I figured out this bug again. I tested on a real device a lot of time and figured. Follow these steps to reproduce ->

- Tap a story
- Snap will play
- Tap on it and long press for 2-3 seconds
- Swipe left and do like pulling out next snap but turn back to current snap

Check the video
https://drive.google.com/file/d/19LmwE-yg-BNV8t49Kxs-dJX4mHmvN29R/view?usp=drivesdk

**Comment by UtkuDalmaz** [comment]
@ranmyfriend could you check it?

**Comment by boominadhaprakash** [comment]
Hi @UtkuDalmaz 

Thanks for raising the issue and your patience. We have fixed the issue and take the source code from the branch: **Boomi/issue89**

Let us know, if your issue got resolved or not from that branch code.

**Comment by UtkuDalmaz** [comment]
Unfortunately, it still stops after swipe animation check the video @boominadhaprakash 

https://drive.google.com/file/d/1PZYdylMcLw8pxjuIgFY6qxk8rF8d8ghf/view?usp=drivesdk

**Comment by boominadhaprakash** [comment]
Hi @UtkuDalmaz 

I have seen you demo video. We are able to reproduce this issue. We will fix this issue at earliest. Thank you.

**Comment by boominadhaprakash** [comment]
Hi @UtkuDalmaz 

I guess, I have fixed this issue and pushed the changes to Boomi/issue89. I have done some basic testing. It is working fine. I will do thorough testing tomorrow. Mean while you also take the changes and please do testing from your side.

I have done changes in IGStoryPreviewCell and IGStoryPreviewController. Please refer the changes and modify your code according to our changes. **Branch: Boomi/issue89**

Let us know if it works fine or not.

**Comment by UtkuDalmaz** [comment]
Thanks, it seems fixed now

**Comment by UtkuDalmaz** [comment]
I realized that now progress doesn’t resume after long press

**Comment by boominadhaprakash** [comment]
Hi @UtkuDalmaz 

Sorry for missed that scenario. We have fixed that issue now and pushed the changes to **Boomi/issue89**. Please take the latest changes of IGStoryPreviewCell and IGStoryPreviewController from **Boomi/issue89** branch.

We have tested the below three scenarios in the latest update.

1. Do Long Press and release, snap progressor should resume.
2. Do Long press, swipe to next story, without releasing the long press, and come back to previous story, snap progressor should resume.
3. Just do normal swipe to next story, without releasing it, come back to previous story. Snap Progressor state should resume.

Let us know if you found issues in existing scenarios or any other scenarios.

**Comment by UtkuDalmaz** [comment]
Ok, this one looks fixed. Now there is another issue. When I long press 2. time it doesn't resume, it goes back or forward. I can long-press and pause it only once. It should pause anytime I long-press just like instagram

**Comment by boominadhaprakash** [comment]
Hi @UtkuDalmaz 

I think, it happened because of simultaneous gesture enabled true. So, we disabled the simultaneous gesture and pushed the code to **Boomi/issue89** branch.

Please check and let us know if you face any issues.

**Comment by UtkuDalmaz** [comment]
Perfect now thanks
